packages/compose/src/hooks/use-debounce/index.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+import { useMemoOne } from 'use-memo-one';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Debounces a function with Lodash's `debounce`. A new debounced function will
+ * be returned and any scheduled calls cancelled if any of the arguments change,
+ * including the function to debounce, so please wrap functions created on
+ * render in components in `useCallback`.
+ *
+ * @param {...any} args Arguments passed to Lodash's `debounce`.
+ */
+export default function useDebounce( ...args ) {
+	const debounced = useMemoOne( () => debounce( ...args ), args );
```

Comment:
I'm not sure I understand why `useMemo` is not enough?

Reply:
Well, it could have a debounced call randomly cancelled. `useMemoOne` guarantees that will never happen. We're using it in `useSelect` too.